clang/lib/Basic/Targets/XCore.h

```diff
@@ -43,7 +43,8 @@ class LLVM_LIBRARY_VISIBILITY XCoreTargetInfo : public TargetInfo {
   void getTargetDefines(const LangOptions &Opts,
                         MacroBuilder &Builder) const override;
 
-  ArrayRef<Builtin::Info> getTargetBuiltins() const override;
+  std::pair<const llvm::StringTable *, ArrayRef<Builtin::Info>>
```

Comment:
This std::pair is used a million times in this patch, IMO this should either:
a- be its own type to give each name a little more context.
OR
b- at minimum, a type alias.

Reply:
Yeah. I mostly didn't invest in tidying this because I knew it was going to change. ;]

Are you OK with just letting the series clean this up, or would you prefer a type alias in the initial patch?

Reply:
The series fixing it up is fine.  Particularly if it stays in 1 PR, where this will end up getting 'squashed' anyway.